yeast-core/src/helpers/diff.ts

```diff
@@ -56,6 +56,68 @@ interface SpaceDiffData {
 	spaceDiffType: DiffType;
 }
 
+export interface AnchorPathMapping {
+	oldPath: string;
+	newPath?: number;
+	isOrphaned: boolean;
+}
+
+export function mapAnchorPath(anchorPath: string, oldNode: DocumentNode, newNode: DocumentNode): AnchorPathMapping {
+	// Navigate to the target node in old document
+	const oldTargetNode = navigateToNodeByPath(oldNode, Number(anchorPath));
```

Comment:
Two questions
1) I was thinking of anchorPath being an array of numbers, which defines the child index at each level, starting from the root and moving down into child arrays. But here it's being cast from string to number and being used as a child index ( `currentNode.children[anchorPath]`). Can you explain your use of anchorPath here?
2) Here, if no oldTargetNode is found with the anchor path, it's considered an orphaned record and the anchor path is assigned to the old node. But if there is no old node at that path, I don't think that path should be assigned to `oldPath`.